designs/2022-docs-information-architecture-update/README.md

```diff
@@ -0,0 +1,351 @@
+  - NOTE: I think there’s further work that can be done to improve the getting started experience, but do not want to touch the subject as part of the information architecture changes project. Investigate further in **Phase 3: “Use ESLint in Your Project” documentation update** of the documentation update project.
+- [NO CHANGE] Core Concepts
+  - Current page: [Core Concepts](https://eslint.org/docs/latest/user-guide/core-concepts)
+- [RENAME PAGE] Configure
+  - Current page: [Configuring](https://eslint.org/docs/latest/user-guide/configuring/)
+  - [RENAME PAGES] Rename the child pages of Configuring as follows:
+    - [NO CHANGE] Configuration Files (New)
+    - [NO CHANGE] Configuration Files
+    - [RENAME PAGE] Configuring Language Options -> Configure Language Options
+    - [RENAME PAGE] Configuring Rules -> Configure Rules
+    - [RENAME PAGE] Configuring Plugins -> Configure Plugins
+    - [RENAME PAGE] Ignoring Rules -> Ignore Rules
+  - Do not change the “Configuring” page and its children pages. Nicholas mentioned that there’s some major revamping to how configuration files work in ESLint, so let’s leave this subsection alone for now.
+- [RENAME PAGE] Command Line Interface Reference
+  - Current page: [Command Line Interface](https://eslint.org/docs/latest/user-guide/command-line-interface)
+- [RENAME PAGE] Rules Reference
+  - Current page: [Rules](https://eslint.org/docs/latest/rules/)
+- [RENAME PAGE] Formatters Reference
+  - Current page: [Formatters](https://eslint.org/docs/latest/user-guide/formatters/)
+- [NO CHANGE] Integrations
+  - Current page: [Integrations](https://eslint.org/docs/latest/user-guide/integrations)
+- [RENAME PAGE] Migrate to v8.x
+  - Current page: [Migrating to v8.0.0](https://eslint.org/docs/latest/user-guide/migrating-to-8.0.0)
+
+### [REFACTOR SECTION]  Extend ESLint
+
+- [NO CHANGE] Extend ESLint landing page
+  - Current page:  [Developer Guide](https://eslint.org/docs/latest/developer-guide/)
+  - Hidden in sidebar
+- [NEW PAGE] Ways to Extend ESLint
+  - Contain overview type content explaining the various ways that you can extend ESLint:
+    - Create Plugins
+    - Create Custom Rules
+    - Create Custom Formatters
+    - Create Custom Parsers
+    - Shareable Configs
+  - NOTE: Should be added, but not part of the initial IA project. Add in **Phase 4: “Extend ESLint” documentation update** of the documentation update project.
+- [RENAME & REFACTOR PAGE] Create Plugins
+  - Current page: [Working with Plugins](https://eslint.org/docs/latest/developer-guide/working-with-plugins)
+  - As noted below, convert the “Processors in Plugins” section into separate page “Custom Processors
+  - Add overview-style content at the top of the page explaining that this page explains how to bundle together the various parts of a plugin, and then publish it.
+- [RENAME PAGE] Custom Rules
+  - Current page: [Working with Rules](https://eslint.org/docs/latest/developer-guide/working-with-rules)
```

Comment:
thanks for calling this out. i think keeping the page as is for now is OK for the initial information architecture changes.

i would imagine that anyone wanting to contribute to the core rules could figure out what to do. and anyone looking to create a custom rule could ignore the content about adding a core rule.

with that being said, i think it'd make sense as a fast follow item to split up the content between these 2 pages. 

i would like to keep these tasks separate b/c i want to keep the information architecture updates very narrowly focused on *moving and renaming* pages and sections, not creating new content. this is to streamline the review process and commit history.

Reply:
> i would imagine that anyone wanting to contribute to the core rules could figure out what to do. and anyone looking to create a custom rule could ignore the content about adding a core rule.

I agree that both an Extender and a Community Contributor will most likely figure out what's relevant to them when they get to the document (though it would be nice to make this clear at some point, as you suggested).

I am concerned about how a new Community Contributor who wants to create a new core rule or change an existing one will get to the document, because "Extend ESLint > Custom Rules" item doesn't sound applicable to what they want to do.

Reply:
> I am concerned about how a new Community Contributor who wants to create a new core rule or change an existing one will get to the document, because "Extend ESLint > Custom Rules" item doesn't sound applicable to what they want to do.

I'm not too concerned about this. We can very easily include a link from the contributing section that says "Core rules are written in the same way as custom rules with the following exceptions."

Reply:
If we plan to move and rename pages before creating new content, after we move and rename pages as proposed it won't be clear for contributors where to look for information on how to create or change a core rule.

https://github.com/eslint/rfcs/blob/c2641344b8c5f3d44f735182e3801aa380dabe8f/designs/2022-docs-information-architecture-update/README.md?plain=1#L181-L200

Reply:
i am going to do a bit of looking at splitting this page up as part of the IA refactor per your points @mdjermanovic. 

as mentioned above, it was my intention to not do any meaningful content refactors as part of the  the IA update. however, this might be an exception to that principle.

plus we could do the page split before the IA-refactor, and then move those 2 separate pages as part of the IA refactor.

Reply:
see this comment for proposed solution here - https://github.com/eslint/rfcs/pull/97#issuecomment-1304605294

Reply:
> see this comment for proposed solution here - [#97 (comment)](https://github.com/eslint/rfcs/pull/97#issuecomment-1304605294)

The comment seems to have disappeared for some reason. I can see it in my email, but can't find it here and this link doesn't work.

Reply:
yeah i don't see it either. this is strange and frustrating. i spent a fair bit of time coming up with that suggestion, so don't want to rethink and rewrite if possible. can you paste it from your email here, and respond to it? (i didn't find it in mine)

Reply:
This one? Pulled from my email:

> @nzakas and @mdjermanovic,
>
> i think i got the solution for this problem of how to split the rules documentation between core rules for contributors and custom rules for extenders.
>
> what i propose is:
>
> 1.  keep the majority of the rule creation content on the "Custom Rules" page in the Extender documentation section. remove content about creating core rules. note somewhere on this page that if you want to create a core rule, refer to the Core Rules documentation (expanded on in next point).
> 2. create a new "Core Rules" page in the Contributor documentation section. extract all content specifically about creating core rules into this page. This page should, as Nicholas proposed, point to the "Custom Rules" page for most of the rule creation documentation.
> 3. do this work as a fast follow to the initial IA migration. while there could certainly be some confusion about how to work on custom rules between the initial migration and the completion of this work, i don't think it'd be a significant issue for the following reasons:
>
> - i don't think many people are working on core rules at a given moment
> - those people can probably figure out what they need to from the custom rules documentation.
> - if those people run into any issues, they can ask in the Discord, and we can point them in the right direction.
> - the amount of time when this is a potential issue would be pretty short, especially if we prioritize the reviews on the content splitting the pages, so that the changes get moved promptly.

Reply:
Here's the thing: the number of times someone will write a new core rule from scratch is tiny compared to the number of times that someone will create a custom rule. Given that, I think we should optimize for the custom rule use case and extend out from that for core rules. I also look forward to the day when there are no core rules, so also don't want to overly-optimize for that.

That said, I'm 👍 to @bpmutter's suggestion as posted above by @btmills

Reply:
@btmills thank you for finding and posting that!

@mdjermanovic how you feel abt the above suggestion?

Reply:
https://github.com/eslint/rfcs/pull/97#discussion_r1017440912 sounds good to me :+1:

Reply:
added to the Documentation update project as part of the Information Architecture update phase 2 as a fast follow to initial updates.